File: working hours register/level 1/C#/workerService.cs
// WorkerService.cs
public class WorkerService
{
    private readonly WorkerRepository _repo;

    public WorkerService(WorkerRepository repo)
    {
        _repo = repo;
    }

    public string AddWorker(string workerId, string position, int compensation)
    {
        return _repo.AddWorker(workerId, position, compensation);
    }

    public string Register(string workerId, int timestamp)
    {
        var worker = _repo.GetWorker(workerId);
        return worker != null ? worker.Register(timestamp) : "invalid_request";
    }

    public string GetTime(string workerId)
    {
        var worker = _repo.GetWorker(workerId);
        return worker != null ? worker.GetTotalMinutes() : "";
    }
}

File: working hours register/level 1/Python/workerservice.py
class WorkerService:
    def __init__(self, repo: WorkerRepository):
        self.repo = repo

    def add_worker(self, worker_id: str, position: str, compensation: int) -> str:
        return self.repo.add_worker(worker_id, position, compensation)

    def register(self, worker_id: str, timestamp: int) -> str:
        worker = self.repo.get_worker(worker_id)
        if not worker:
            return "invalid_request"
        return worker.register(timestamp)

    def get_time(self, worker_id: str) -> str:
        worker = self.repo.get_worker(worker_id)
        return worker.get_total_minutes() if worker else ""
